torch/optim/sgd.py

```diff
@@ -104,6 +112,9 @@ def step(self, closure=None):
                     else:
                         d_p = buf
 
+                if weight_decay != 0 and not group['l2_reg']:
+                    p.data.add_(-weight_decay, p.data)
```

Comment:
```suggestion
                    p.data.add_(-weight_decay * group['lr'], p.data)
```

In L2, the weight decay is stored inside the grad variable, which is later multiplied by the learning rate. This new formulation is not multiplying it by the learning rate (see Algorithms 1 and 2 in the paper).